### DISCUSSION TITLE
How to Make Certain Fields Optional in Forms?

### DISCUSSION THREAD

**Discussion by Ali-Makke** [horilla-opensource/horilla#337] [General]
I have several forms where all fields are currently required. I want to modify the forms so that I can choose which fields are required and which are optional.

For example, in the Create Company form (screenshot attached), all fields are currently mandatory, but I would like to adjust this. How can I configure each form field to be required or optional based on my needs?
![Screenshot 2024-09-11 174835](https://github.com/user-attachments/assets/f1b7e3f2-57e1-48e0-a7b4-f7d95153e974)

**Comment by horilla-opensource** [comment]
> I have several forms where all fields are currently required. I want to modify the forms so that I can choose which fields are required and which are optional.
> 
> For example, in the Create Company form (screenshot attached), all fields are currently mandatory, but I would like to adjust this. How can I configure each form field to be required or optional based on my needs? ![Screenshot 2024-09-11 174835](https://private-user-images.githubusercontent.com/68686511/366509292-f1b7e3f2-57e1-48e0-a7b4-f7d95153e974.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwNzA4NTMsIm5iZiI6MTcyNjA3MDU1MywicGF0aCI6Ii82ODY4NjUxMS8zNjY1MDkyOTItZjFiN2UzZjItNTdlMS00OGUwLWE3YjQtZjdkOTUxNTNlOTc0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDE2MDIzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY2OWYzNTM2NWI1ZDM4MDc4NDBiZGI4MmJjYTUyYjA2YzNjOGYwYTYxNTNjMjBlMDY3NjgxNTMzMzA4ZWYzOTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.HSnxwWwrxnM9kbBSVUuYgdAAGNq3qD1D-N_qUghAPZs)

Hi @PurpleGhostRider ,
To have the fields required/ not required can only be made from the backend and not frontend as there are certain validations created for the fields in the database table.
To have the above mentioned features, will require customizations on the backend models.

**Reply by Ali-Makke** [reply]
Thank you for your reply!

Could you guide me on where exactly I can make these changes? Specifically:

- Which file or model should I look for?
- What should I edit to make certain fields optional?

Any detailed steps or examples would be greatly appreciated.

**Reply by horilla-opensource** [reply]
> Thank you for your reply!
> 
> Could you guide me on where exactly I can make these changes? Specifically:
> 
> * Which file or model should I look for?
> * What should I edit to make certain fields optional?
> 
> Any detailed steps or examples would be greatly appreciated.

Hi @PurpleGhostRider ,
Which all areas do you need the customisation required?

**Reply by Ali-Makke** [reply]
Hello,
I was able to remove the required fields that I didn’t need. Thanks again for your response and willingness to help!!
